### PR TITLE
allow calling [github] without auth

### PR DESCRIPTION
**Pull request by chris48s**
Closes https://github.com/badges/shields/issues/2754
Refs https://github.com/badges/shields/pull/9386#discussion_r1264705287

This PR resolves a long-standing issue that even though GitHub allows anonymous access to the v3 API with a reasonable rate limit, you have to set a key locally to do any development on any GitHub badges. The changes in https://github.com/badges/shields/pull/9386 make it extra desirable to be able to call the GH API without auth if possible.

I've set this PR to run the github test suite. Five of them will fail, but I submitted another PR to fix the tests at https://github.com/badges/shields/pull/9425 so we could merge that first then rebase this onto it, although I think the integration tests are actually the critical ones here.

### CODE REVIEW

File: services/github/github-auth-service.js [hunk 1]
```diff
@@ -63,7 +63,14 @@ class GithubAuthV4Service extends BaseGraphqlService {
       `,
     )
 
-    return super._requestGraphql({ ...attrs, ...{ url, query } })
+    return super._requestGraphql({
+      ...attrs,
+      ...{
+        url,
+        query,
+        httpErrorMessages: { 401: 'auth required for graphql api' },
```
Comment (chris48s) [inline services/github/github-auth-service.js:71]: The v4/GraphQL API doesn't allow anonymous access - a token is required

File: services/github/github-search.service.js [hunk 2]
```diff
@@ -49,7 +49,11 @@ export default class GithubSearch extends GithubAuthV3Service {
         },
       },
       schema,
-      httpErrors: httpErrorsFor('repo not found'),
+      httpErrors: {
+        401: 'auth required for search api',
```
Comment (chris48s) [inline services/github/github-search.service.js:53]: The search API doesn't allow anonymous access - a token is required

File: services/github/github-api-provider.integration.js [hunk 3]
```diff
@@ -30,7 +51,12 @@ describe('Github API provider', function () {
     it('should be able to run 10 requests', async function () {
       this.timeout('20s')
       for (let i = 0; i < 10; ++i) {
-        await githubApiProvider.fetch(fetch, '/repos/rust-lang/rust', {})
+        const { res } = await githubApiProvider.fetch(
+          fetch,
+          '/repos/rust-lang/rust',
+          {},
+        )
+        expect(res.statusCode).to.equal(200)
```
Comment (chris48s) [inline services/github/github-api-provider.integration.js:59]: When I was working on this, I realised this test never actually asserted the calls were successful! :grimacing:

**Review by PyvesB — APPROVED** [review]
Changes look good to me 👍🏻

**Pull request merged**